pilot/pkg/leaderelection/leaderelection_test.go

```diff
@@ -127,6 +136,38 @@ func TestPrioritizedLeaderElection(t *testing.T) {
 	close(stop7)
 }
 
+func TestMulticlusterLeaderElection(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	watcher := &fakeDefaultWatcher{}
+	// First remote pod becomes the leader
+	_, stop := createElectionMulticluster(t, "pod1", "", true, watcher, false, true, client)
+	// A new local pod takes over as leader
+	_, stop2 := createElectionMulticluster(t, "pod2", "", false, watcher, false, true, client)
+	// A new remote pod cannot become leader
+	_, stop3 := createElectionMulticluster(t, "pod3", "", true, watcher, false, false, client)
+	close(stop3)
+	close(stop2)
+	close(stop)
+}
+
+func TestPrioritizedMulticlusterLeaderElection(t *testing.T) {
+	client := fake.NewSimpleClientset()
```

Comment:
use `kube.NewFakeClient` - this may have test flakes because the k8s fake client is not safe by default

Reply:
I'm not following what you mean. I just copied this from the other tests.

Are you suggesting I change something here?

Reply:
One of the tests calls `client.Fake.PrependReactor` which doesn't work with the `ExtendedClient` returned by `NewFakeClient()`, so I'll just leave all of them as `NewSimpleClientset()` which seems to work fine.